File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/analytics/model/ToggleLanguageAnalyticsActionEvent.java
package org.cru.godtools.tract.analytics.model;

import org.ccci.gto.android.common.compat.util.LocaleCompat;
import org.cru.godtools.analytics.model.AnalyticsActionEvent;
import org.cru.godtools.analytics.model.AnalyticsSystem;

import java.util.Collections;
import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import static org.cru.godtools.analytics.AdobeAnalyticsService.KEY_CONTENT_LANGUAGE_SECONDARY;

public final class ToggleLanguageAnalyticsActionEvent extends AnalyticsActionEvent {
    private static final String ACTION_TOGGLE_LANGUAGE = "Parallel Language Toggled";
    private static final String KEY_ADOBE_TOGGLE_LANGUAGE = "cru.parallellanguagetoggle";

    private final String mTract;
    private final Map<String, ?> mAttrs;

    public ToggleLanguageAnalyticsActionEvent(@Nullable final String tract, @NonNull final Locale locale) {
        super(null, ACTION_TOGGLE_LANGUAGE);
        mTract = tract;
        final HashMap<String, Object> attrs = new HashMap<>();
        attrs.put(KEY_ADOBE_TOGGLE_LANGUAGE, 1);
        attrs.put(KEY_CONTENT_LANGUAGE_SECONDARY, LocaleCompat.toLanguageTag(locale));
        mAttrs = Collections.unmodifiableMap(attrs);
    }

    @Override
    public boolean isForSystem(@NonNull final AnalyticsSystem system) {
        return system == AnalyticsSystem.ADOBE || system == AnalyticsSystem.FACEBOOK;
    }

    @Nullable
    @Override
    public Map<String, ?> getAttributes() {
        return mAttrs;
    }

    @Nullable
    @Override
    public String getAdobeSiteSection() {
        return mTract;
    }
}

File: ui/tract-renderer/src/main/java/org/cru/godtools/tract/analytics/model/ToggleLanguageAnalyticsActionEvent.kt
package org.cru.godtools.tract.analytics.model

import android.os.Bundle
import java.util.Locale
import org.ccci.gto.android.common.compat.util.LocaleCompat.toLanguageTag
import org.cru.godtools.analytics.firebase.PARAM_LANGUAGE_SECONDARY
import org.cru.godtools.analytics.model.AnalyticsActionEvent
import org.cru.godtools.analytics.model.AnalyticsSystem

private const val ACTION_TOGGLE_LANGUAGE = "parallel_language_toggled"

class ToggleLanguageAnalyticsActionEvent(tool: String?, private val selectedLocale: Locale) :
    AnalyticsActionEvent(action = ACTION_TOGGLE_LANGUAGE) {
    override fun isForSystem(system: AnalyticsSystem) =
        system === AnalyticsSystem.FIREBASE || system === AnalyticsSystem.FACEBOOK

    override val appSection = tool

    override val firebaseParams get() = Bundle().apply {
        putString(PARAM_LANGUAGE_SECONDARY, toLanguageTag(selectedLocale))
    }
}
